larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -52,6 +52,15 @@ inline void decide_bitpack_before_im2col(KernelType kernel_type,
   }
 }
 
+#define LCE_ENSURE_PARAM(conv_params, context, a)                       \
```

Comment:
I didn't use `TFLITE_ENSURE` here since it would return `kTfliteError` instead of `conv_params`. Not sure if this is the best way to handle the errors though.

Reply:
This looks like a nice solution to me.

Side note: There is no solution we can copy from tflite, because they don't have to read a flexbuffer in `Init`: their `struct` with parameters is already filled, and correctness is ensured by the Interpreter and the flatbuffer schema. Therefore they never have to report errors from `Init`.

Reply:
We could have the following to speed up non-test builds, but since its only for initialization I think we don't have to.
```cpp
#ifdef NDEBUG
#define LCE_ENSURE_PARAM(conv_params, context, a) 
#else
#define LCE_ENSURE_PARAM(conv_params, context, a)  do { if(!(a)) { ... } } while(0)
#endif
```

Reply:
Thanks for the explanation.

> We could have the following to speed up non-test builds, but since its only for initialization I think we don't have to.

I don't think we need to do this, since `TF_LITE_ENSURE` uses a very similar macro which is used often throughout the codebase so optimizing compile time of `LCE_ENSURE_PARAM` will probably not be noticable.

On a different note, if we really need to reduce binary size, we can replace calls to `context->ReportError` with [`TF_LITE_KERNEL_LOG`](https://github.com/tensorflow/tensorflow/blob/f44bb87843cababb97401e93ae9d91e6a296c293/tensorflow/lite/c/common.h#L153-L171) which allows to strip out message strings if the `TF_LITE_STRIP_ERROR_STRINGS` flag is set.

Reply:
Agreed. And that `TF_LITE_KERNEL_LOG` is a good idea, lets do that in a separate PR.